modules/standard/IO.chpl

```diff
@@ -4398,6 +4401,19 @@ proc channel.close() {
   if e then this._ch_ioerror(e, "in channel.close");
 }
 
+// TODO -- we should probably have separate c_ptr ddata and ref versions
```

Comment:
Not sure why "but" is here. Don't we want both a better comment and compile-time checking?

Having C compilation failures or core dumps is not cool. Can you add compile-time checking within a reasonable amount of effort?